tools/generators/release-package/index.ts

```diff
@@ -0,0 +1,328 @@
+import {
+  Tree,
+  formatFiles,
+  names,
+  updateJson,
+  ProjectConfiguration,
+  joinPathFragments,
+  visitNotIgnoredFiles,
+  createProjectGraphAsync,
+  reverse,
+  installPackagesTask,
+  readJson,
+  stripIndents,
+} from '@nrwl/devkit';
+
+import * as tsquery from '@phenomnomnominal/tsquery';
+
+import { getProjectConfig, workspacePaths } from '../../utils';
+
+import { PackageJson, TsConfig } from '../../types';
+
+import tsConfigBaseAll from '../tsconfig-base-all';
+
+import { ReleasePackageGeneratorSchema } from './schema';
+import { execSync } from 'child_process';
+
+interface NormalizedSchema extends ReturnType<typeof normalizeOptions> {}
+
+export default async function (tree: Tree, schema: ReleasePackageGeneratorSchema) {
+  const options = normalizeOptions(tree, schema);
+
+  assertProject(tree, options);
+
+  const tasks: Array<(tree: Tree) => void> = [];
+
+  if (options.phase === 'initial') {
+    tasks.push(initialRelease(tree, options));
+  }
+  if (options.phase === 'stable') {
+    tasks.push(await stableRelease(tree, options));
+  }
+
+  await formatFiles(tree);
+
+  return () => {
+    tasks.forEach(task => {
+      task(tree);
+    });
+  };
+}
+
+function normalizeOptions(tree: Tree, options: ReleasePackageGeneratorSchema) {
+  const project = getProjectConfig(tree, { packageName: options.project });
+
+  return {
+    ...options,
+    ...project,
+    ...names(options.project),
+  };
+}
+
+function initialRelease(tree: Tree, options: NormalizedSchema) {
+  updateJson<PackageJson>(tree, options.paths.packageJson, json => {
+    delete json.private;
+    return json;
+  });
+
+  const docsiteProjectName = '@' + options.workspaceConfig.npmScope + '/public-docsite-v9';
+  const docsite = getProjectConfig(tree, { packageName: docsiteProjectName });
+
+  updateJson<PackageJson>(tree, docsite.paths.packageJson, json => {
+    json.dependencies = json.dependencies ?? {};
```

Comment:
```suggestion
    json.dependencies ??= {};
```

Reply:
from experience I tend to avoid "javascript magic"  (in this case short-circuit assigments) - extremely crytpic to human brain

Reply:
it's not just javascript though https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator